niceman/resource/tests/test_singularity.py

```diff
@@ -0,0 +1,81 @@
+# emacs: -*- mode: python; py-indent-offset: 4; tab-width: 4; indent-tabs-mode: nil -*-
+# ex: set sts=4 ts=4 sw=4 noet:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the niceman package for the
+#   copyright and license terms.
+#
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+
+
+# NOTE: The singularity classes SingularitySession and PTYSingularitySession
+# are tested in test_session.test_session_abstract_methods()
+
+import os
+import tempfile
+
+import logging
+import re
+from ..singularity import Singularity, SingularitySession
+from ...tests.utils import skip_if_no_singularity, skip_if_no_network, \
+    swallow_logs, assert_in
+
+
+@skip_if_no_network
+@skip_if_no_singularity
+def test_singularity_resource_class():
+
+    with swallow_logs(new_level=logging.DEBUG) as log:
+
+        # Set working directory to a scratch directory since we will be creating
+        # Singularity image files during testing.
+        tempdir = tempfile.mkdtemp()
+        os.chdir(tempdir)
```

Comment:
Use `with chpwd(tempdir)` context manager here to make sure you come back "correctly".  This is what caused the test failure of `test_assert_Xwd_unchanged`